### PR TITLE
Feature/sequential refactoring

### PR DESCRIPTION
**Pull request by dimarkov**
This PR does a bit of refactoring, to reduce the number of scripts we need to run different model variants. 

I have also introduce a sequential training scheme for full network fine tuning, where we iterate twice through the dataset, first to train the last layer, and then with a frozen last layer train the neural network via Bayesian loss.

Finally, I have added a reset last layer feature to the training regime, so that after each epoch the last layer gets re-initialized (only in the case of IBProbit loss).

### CODE REVIEW

**Review by conorheins — APPROVED** [review]
Looks good to me, just added a warning to the # validate arguments block that when using IBProbit and last layer finetuning, the optimizer/opt_state/relevant hyperparameters for the chosen optimizer, will be ignored. I also changed the string of the default wandb group to be `finetuning_test` away from the old group id.
